source/slang/ir.cpp

```diff
@@ -4601,7 +4579,8 @@ namespace Slang
         // At this point we've created a new non-generic function,
         // which means we should add it to our work list for
         // subsequent processing.
-        sharedContext->workList.Add(specFunc);
+        if (specFunc->specializedGenericLevel == -1)
+            sharedContext->workList.Add(specFunc);
```

Comment:
This is extremely minor, but I would have assumed that a `grnericSpecializationLevel` of zero would represent the non-generic case, not `-1`.

Reply:
Yeah, but `genericSpecializationLevel` is used as an index into the `genericDecls` list, where the last element is the outer most decl and first element is inner most generic decl. So a non-generic function will have an empty `genericDecls` list and `genericSpecializationLevel==-1`.